pkg/gitparse/gitparse.go

```diff
@@ -681,27 +681,33 @@ func isBinaryLine(latestState ParseState, line []byte) bool {
 }
 
 // Get the b/ file path. Ignoring the edge case of files having `and /b` in the name for simplicity.
-func pathFromBinaryLine(line []byte) (string, bool) {
+func pathFromBinaryLine(ctx context.Context, line []byte) (string, bool) {
 	if bytes.Contains(line, []byte("and /dev/null")) {
 		return "", true
 	}
 
-	_, after, ok := bytes.Cut(line, []byte(" and b/"))
-	if ok {
+	var path string
+	if _, after, ok := bytes.Cut(line, []byte(" and b/")); ok {
 		// drop the " differ\n"
-		return string(after[:len(after)-8]), true
-	}
+		path = string(after[:len(after)-8])
+	} else if _, after, ok = bytes.Cut(line, []byte(` and "b/`)); ok {
+		// Edge case where the path is quoted.
+		// https://github.com/trufflesecurity/trufflehog/issues/2384
 
-	// Edge case where the path is quoted.
-	// https://github.com/trufflesecurity/trufflehog/issues/2384
-	_, after, ok = bytes.Cut(line, []byte(` and "b/`))
-	if ok {
 		// drop the `" differ\n`
-		return string(after[:len(after)-9]), true
+		path = string(after[:len(after)-9])
+	} else {
+		// Unknown format.
+		return "", false
+	}
+
+	unicodePath, err := strconv.Unquote(`"` + path + `"`)
```

Comment:
What made you use this technique over sprintf? Either way, would you mind adding a comment explaining the pitfall you're now avoiding?

Reply:
question: Do we need the string concat, could we just pass in `path`, because it seems that the path variable is expected to be either an unquoted string or a double-quoted string?

It's also very likely i'm just missing something obvious here 😓

Reply:
> What made you use this technique over sprintf? Either way, would you mind adding a comment explaining the pitfall you're now avoiding?

That was speculation given that `fmt.Printf` seemed to worked; I checked, and sprintf doesn't work.

> question: Do we need the string concat, could we just pass in path, because it seems that the path variable is expected to be either an unquoted string or a double-quoted string?

[The quotes are required](https://pkg.go.dev/strconv#Unquote), for some reason.

Reply:
fwiw when joining strings with few arguments using `+` is generally the fastest approach as well.
Benchmarks with a few different approaches:
[some benchmarks](https://gist.github.com/dtjm/c6ebc86abe7515c988ec)